data/gencode/convert.gencode.genes.to.tss.py

```diff
@@ -0,0 +1,23 @@
+import gzip
+import json
+
+# Extract transcript start site information from the file containing GENCODE gene information
+# Also encodes the strand information as the sign of the TSS:
+# a TSS on the + strand is recorded as a positive TSS, while
+# a TSS on the - strand is recorded as a negative TSS
+tssDict = dict()
+with gzip.open('gencode.v30.annotation.gtf.genes.bed.gz', 'r') as f:
+    for line in f:
+        temp = line.decode('utf-8').rstrip('\n').split()
+        if len(temp) == 9:
+            (chrom, dataSource, dataType, start, end, strand, geneID, geneType, geneSymbol) = temp
+            if strand == '+':
+                tss = int(start)
+            elif strand == '-':
+                tss = -1 * int(end)
+            else:
+                tss = -1
```

Comment:
Under what circumstances will GENCODE not provide a strand?

Does the value -1 have special meaning, other than as a placeholder for "no value available"? JSON supports `null` (or as would be said in Python, `None`), if we wanted a more intuitive placeholder.

Reply:
Reviewing this, I think the proper action is to use None here, and if the strand information is not present, don't write anything to the dictionary at all. I've reviewed the latest version (v38) of the gencode data and every single entry has a strand designation (+ or -).